AEPIdentity/Sources/Identity.swift

```diff
@@ -55,41 +55,94 @@ import Foundation
+        let isConfigSharedStateSet = getSharedState(extensionName: IdentityConstants.SharedStateKeys.CONFIGURATION, event: event)?.status == .set
+
+        if !isConfigSharedStateSet {
+            Log.trace(label: "\(name):\(#function)", "Waiting for the Configuration shared state to be set before processing [event:(\(event.name)) id:(\(event.id)].")
+        }
+
+        return isConfigSharedStateSet
     }
 
-    /// Determines if Identity is ready to handle events, this is determined by if the Identity extension has booted up
+    // Loads all the identifiers in identity state from persistence, generates ecid if not found in persistence and gets the extension ready for get identifiers event.
     /// - Parameter event: An `Event`
     /// - Returns: True if we can process events, false otherwise
-    private func canProcessEvents(event: Event) -> Bool {
+    private func bootUp(event: Event) -> Bool {
         guard let state = state else { return false }
         guard !state.hasBooted else { return true } // we have booted, return true
 
-        // fetch latest configuration when booting up
-        guard let configSharedState = getSharedState(extensionName: IdentityConstants.SharedStateKeys.CONFIGURATION, event: nil)?.value else { return false }
         // attempt to bootup
-        if state.bootupIfReady(configSharedState: configSharedState, event: event) {
+        if state.boot(event: event) {
             createSharedState(data: state.identityProperties.toEventData(), event: nil)
+            return true
+        }
+
+        return false // cannot handle any events until we have booted
+
+    }
+
+    /// Determines if Identity is ready to handle events, this is determined by if the Identity extension has received all the required configuration
+    /// - Parameter event: An `Event`
+    /// - Returns: True if we can process events, false otherwise
+    private func canProcessEvents(event: Event) -> Bool {
+        guard let state = state else { return false }
+        guard !state.hasSynced else { return true } // we have synced, return true
+
+        // fetch latest configuration when booting up
+        guard let configSharedState = getSharedState(extensionName: IdentityConstants.SharedStateKeys.CONFIGURATION, event: nil)?.value else {
+            Log.trace(label: "\(name):\(#function)", "Waiting for the Configuration shared state to get required configuration fields before processing [event:(\(event.name)) id:(\(event.id)].")
+            return false
+        }
+
+        if state.SyncIdentifiersIfConfigurationAvailable(configSharedState: configSharedState, event: event) {
+            return true
         }
 
+        Log.trace(label: "\(name):\(#function)", "Waiting for the Identity state to Sync up. Identity will sync up when it has recevied all the required configuration before processing [event:(\(event.name)) id:(\(event.id)].")
+
         return false // cannot handle any events until we have booted
+    }
```

Comment:
return state.SyncIdentifiersIfConfigurationAvailable(configSharedState: configSharedState, event: event)

Reply:
I wanted to have log for false case.